module-sys/Service/Service.cpp

```diff
@@ -112,6 +112,7 @@ namespace sys
                                  staleUniqueMsg.end());
 
             const bool respond = msg->type != Message::Type::Response && GetName() != msg->sender;
+            currentlyProcessing = msg;
```

Comment:
maybe set back to null if nothing processed currently

Reply:
no, this would be wrong